integration/spark/src/main/java/openlineage/spark/agent/lifecycle/plan/PlanUtils.java

```diff
@@ -92,100 +92,107 @@ public static String namespaceUri(URI outputPath) {
+   * Construct a {@link OpenLineage.DatasetFacets} given a schema, a namespace, and an {@link
    * OutputStatisticsFacet}.
    *
    * @param schema
    * @param namespaceUri
    * @param outputStats
    * @return
    */
-  public static LineageEvent.DatasetFacet datasetFacet(
+  public static OpenLineage.DatasetFacets datasetFacet(
       StructType schema, String namespaceUri, OutputStatisticsFacet outputStats) {
-    return LineageEvent.DatasetFacet.builder()
-        .schema(schemaFacet(schema))
-        .dataSource(datasourceFacet(namespaceUri))
-        .additional(ImmutableMap.of("stats", outputStats))
-        .build();
+    OpenLineage.DatasetFacetsBuilder builder =
+        new OpenLineage.DatasetFacetsBuilder()
+            .schema(schemaFacet(schema))
+            .dataSource(datasourceFacet(namespaceUri));
+
+    builder.put("stats", outputStats);
+    return builder.build();
   }
 
   /**
-   * Construct a {@link LineageEvent.DatasourceDatasetFacet} given a namespace for the datasource.
+   * Construct a {@link OpenLineage.DatasourceDatasetFacet} given a namespace for the datasource.
    *
    * @param namespaceUri
    * @return
    */
-  public static LineageEvent.DatasourceDatasetFacet datasourceFacet(String namespaceUri) {
-    return LineageEvent.DatasourceDatasetFacet.builder()
-        ._producer(URI.create(OpenLineageClient.OPEN_LINEAGE_CLIENT_URI))
-        ._schemaURL(URI.create(OpenLineageClient.OPEN_LINEAGE_DATASOURCE_FACET))
-        .uri(namespaceUri)
+  public static OpenLineage.DatasourceDatasetFacet datasourceFacet(String namespaceUri) {
+    return new OpenLineage(URI.create(OpenLineageClient.OPEN_LINEAGE_CLIENT_URI))
+        .newDatasourceDatasetFacetBuilder()
+        .uri(URI.create(namespaceUri))
         .name(namespaceUri)
         .build();
   }
 
   /**
-   * Construct a {@link LineageEvent.ParentRunFacet} given the parent job's runId, job name, and
+   * Construct a {@link OpenLineage.ParentRunFacet} given the parent job's runId, job name, and
    * namespace.
    *
    * @param runId
    * @param parentJob
    * @param parentJobNamespace
    * @return
    */
-  public static LineageEvent.ParentRunFacet parentRunFacet(
+  public static OpenLineage.ParentRunFacet parentRunFacet(
       String runId, String parentJob, String parentJobNamespace) {
-    return LineageEvent.ParentRunFacet.builder()
-        ._producer(URI.create(OpenLineageClient.OPEN_LINEAGE_CLIENT_URI))
-        ._schemaURL(URI.create(OpenLineageClient.OPEN_LINEAGE_PARENT_FACET_URI))
-        .run(LineageEvent.RunLink.builder().runId(runId).build())
-        .job(LineageEvent.JobLink.builder().name(parentJob).namespace(parentJobNamespace).build())
+    return new OpenLineage(URI.create(OpenLineageClient.OPEN_LINEAGE_CLIENT_URI))
+        .newParentRunFacetBuilder()
+        .run(
+            new OpenLineage.ParentRunFacetRunBuilder()
+                .runId(convertToUUID.apply(runId).orElse(UUID.randomUUID()))
```

Comment:
If there's no parent runUUID, don't just make one up. The `ParentRunFacet` should just be omitted if there is no valid run id.